.github/workflows/azure-e2e-test.yml

```diff
@@ -93,10 +96,14 @@ jobs:
     steps:
     - uses: actions/checkout@v3
 
+    - name: Extract go version number
+      run: echo "GO_VERSION=$(yq -e '.tools.golang' versions.yaml)" >> "$GITHUB_ENV"
```

Comment:
Doesn't GITHUB_ENV work across the whole job? As in do we need to do this extraction twice?

Reply:
I don't think it can. Jobs run on different machines, there is no way this could work without being super racy. I think you can define a global ENV, but that must be static.